sql/sql_delete.cc

```diff
@@ -312,15 +312,15 @@ bool mysql_delete(THD *thd, TABLE_LIST *table_list, COND *conds,
 
   THD_STAGE_INFO(thd, stage_init_update);
 
-  if (table_list->period_conditions.is_set() && table_list->vers_conditions.is_set())
-  {
-    my_error(ER_SYNTAX_ERROR, MYF(0));
-    DBUG_RETURN(true);
-  }
-
   bool truncate_history= table_list->vers_conditions.is_set();
   if (truncate_history)
   {
+    if (table_list->period_conditions.is_set())
+    {
+      my_error(ER_SYNTAX_ERROR, MYF(0));
+      DBUG_RETURN(true);
+    }
+
```

Comment:
These things looks trivial to me. There is nothing to try to be more clear because it's clear the either way. What I like here is single condition just because it's shorter. Nesting level doesn't matter when it isn't more than 4.

Reply:
Looks like a good rule of thumb. I'd prefer, maybe, 3, not 4. Or 2.